griptape/drivers/vector/redis_vector_store_driver.py

```diff
@@ -120,8 +123,9 @@ def query(
 
         vector = self.embedding_driver.embed_string(query)
 
+        filter_expression = f"(@namespace:{{{namespace}}})" if namespace else "*"
```

Comment:
If the `namespace` does not exist in the index schema, then passing `namespace` to query will never find any results, including when it belongs to a tool.

So in my example above, `driver.query("What is griptape?", namespace="griptape")` would produce 0 results instead of 1 and the `VectorStoreClient` tool (configured with a `namespace`) will never find any results.

I think we might be able to inspect the schema to either change this behavior or provide a warning to prevent people from shooting their feet. Not every query, on initialization. I'll see what I can do. Let me know if you have other ideas.

Reply:
Gotcha. So, when no `namespace` parameter is specified and we add the `@namespace:*` filter, will it return empty results as well? If so, we should probably not include that filter, right?

Reply:
If no `namespace` is provided, `filter_expression` will evaluate to `'*'` rather than `@namespace:*` (there's a ternary). Providing no `namespace` will search through all vectors in the index just as before. 

This is illustrated in the example program via `driver.query("What is griptape?")`, which returns a result.

Reply:
Oops, I misread the code, sorry! Let's ship it!